src/java.base/share/classes/java/security/FIPSConfigurator.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * ===========================================================================
+ * (c) Copyright IBM Corp. 2022, 2022 All Rights Reserved
+ * ===========================================================================
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * IBM designates this particular file as subject to the "Classpath" exception
+ * as provided by IBM in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, see <http://www.gnu.org/licenses/>.
+ *
+ * ===========================================================================
+ */
+
+package java.security;
+
+import java.util.Iterator;
+import java.util.Map.Entry;
+import java.util.Properties;
+import java.util.List;
+
+import sun.security.util.Debug;
+
+/**
+ * Configures the security providers when in FIPS mode.
+ */
+public final class FIPSConfigurator {
+
+    static final Debug debug = Debug.getInstance("semerufips");
+
+    // FIPS mode enable check, only support on Linux x64
+    private static boolean userEnabledFIPS = false;
+    private static boolean isFIPSSupported = false;
+    private static boolean shouldEnableFIPS = false;
+
+    static {
+        List<String> props = AccessController.doPrivileged(
+                new PrivilegedAction<>() {
+                    @Override
+                    public List<String> run() {
+                        return List.of(
+                                System.getProperty("semeru.fips", "false"),
+                                System.getProperty("os.name"),
+                                System.getProperty("os.arch"));
+                    }
+                });
+        userEnabledFIPS = Boolean.valueOf(props.get(0));
+        isFIPSSupported = "Linux".equalsIgnoreCase(props.get(1))
+                && "amd64".equalsIgnoreCase(props.get(2));
+        shouldEnableFIPS = userEnabledFIPS && isFIPSSupported;
+    }
+
+    /**
+     * FIPS mode will be enabled only if the semeru.fips system
+     * property is true (default as false)
```

Comment:
Please finish each sentence with a period.

Reply:
Please complete the javadoc to avoid this warning:
```
support/j9jcl/java.base/share/classes/java/security/FIPSConfigurator.java:66: warning: no @return
    public static boolean enableFips() {
                          ^
```